packages/labs/compiler/src/lib/template-transform.ts

```diff
@@ -343,6 +343,20 @@ class CompiledTemplatePass {
               type: PartType.CHILD,
               index: nodeIndex,
             });
+          } else {
+            // Handle a dynamic binding within a comment node, e.g.:
+            // html`<!--<div>${'binding'}</div>--><p>${'hi'}</p>`
+            let i = -1;
+            while ((i = node.data.indexOf(marker, i + 1)) !== -1) {
+              // Comment node has a binding marker inside, make an inactive part
+              // The binding won't work, but subsequent bindings will
+              parts.push({type: PartType.COMMENT_PART, index: nodeIndex});
+              // Move to the end of the match
+              i += marker.length - 1;
+            }
+            // In the compiled result, remove the markers, so compiled
+            // files are deterministic.
+            node.data = node.data.split(marker).join('');
```

Comment:
nit: what do you think about doing 
```suggestion
            node.data.replaceAll(marker, '');
```

Reply:
I think that's a far superior method.

Reply:
Oh no: Property 'replaceAll' does not exist on type 'string'. Do you need to change your target library? Try changing the 'lib' compiler option to 'es2021' or later.

Reply:
Fixed

Reply:
Darn. Too new. I suppose. Since we're moving to es2021 for 3.0 maybe that's okay to bump. Otherwise split and join, or we'd need to `.replace()` constructing a RegExp with the marker, escaping the `$`.